src/GrumPHP/Task/PHPUnitCoverageTask.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Configuration\GrumPHP;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use GrumPHP\Task\TaskInterface;
+use InvalidArgumentException;
+use SimpleXMLElement;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class PHPUnitCoverageTask implements TaskInterface
+{
+    /**
+     * @var GrumPHP
+     */
+    protected $grumPHP;
+
+    /**
+     * @param GrumPHP $grumPHP
+     */
+    public function __construct(GrumPHP $grumPHP)
+    {
+        $this->grumPHP = $grumPHP;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfiguration()
+    {
+        $configured = $this->grumPHP->getTaskConfiguration($this->getName());
+
+        return $this->getConfigurableOptions()->resolve($configured);
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return 'phpunit_coverage';
+    }
+
+    /**
+     * @return OptionsResolver
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'clover_file' => null,
+            'level' => 100,
+        ]);
+
+        $resolver->addAllowedTypes('clover_file', ['string']);
```

Comment:
If clover file is required and doesn't have a default value, you can mark this property as required and remove the default value. (Since this value is null and the allowed types are expecting a string.)